### PR TITLE
feat(3056): add endpoint to mark pipeline template as trusted

### PR DESCRIPTION
**Pull request by klu909**
## Context

add endpoint to mark pipeline template as trusted
## Objective

add endpoint to mark pipeline template as trusted
## References
https://github.com/screwdriver-cd/screwdriver/issues/3056

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by tkyi — CHANGES_REQUESTED** [review]
Can you update README: https://github.com/screwdriver-cd/screwdriver/blob/master/plugins/pipelines/README.md

File: plugins/pipelines/templates/updateTrusted.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        tags: ['api', 'pipeline', 'templates', 'trusted'],
+        auth: {
+            strategies: ['token'],
+            scope: ['admin', '!guest']
+        },
+
+        handler: async (request, h) => {
+            const { namespace, name } = request.params;
+            const { pipelineTemplateFactory } = request.server.app;
+            const { trusted } = request.payload;
```
Comment (pritamstyz4ever) [inline plugins/pipelines/templates/updateTrusted.js:23]: the trusted API should also take `version` in the payload, to mark specific versions and lower as trusted

File: plugins/pipelines/templates/updateTrusted.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            const pipelineTemplateMeta = await pipelineTemplateFactory.get({
+                name,
+                namespace
+            });
+
+            if (!pipelineTemplateMeta) {
+                throw boom.notFound(`Pipeline template ${namespace}/${name} does not exist`);
+            }
+
+            if (!trusted) {
```
Comment (pritamstyz4ever) [inline plugins/pipelines/templates/updateTrusted.js:34]: if no `version` is specified then use the latest, else update with the version from payload

**Review by pritamstyz4ever — CHANGES_REQUESTED** [review]

File: plugins/pipelines/templates/updateTrusted.js (hunk 2 above)
Comment (klu909) [reply plugins/pipelines/templates/updateTrusted.js:34]: resolving this as per discussion in slack

**Review by pritamstyz4ever — APPROVED** [review]
Let's stick to marking latest as trusted as it will be done by SD Admin

**Review by tkyi — APPROVED** [review]

**Review by tkyi — CHANGES_REQUESTED** [review]
Can you add a test for when `trusted: false`

**Review by tkyi — APPROVED** [review]

**Pull request merged**